dds/DCPS/unique_ptr.h

```diff
@@ -69,26 +90,36 @@ class scoped_ptr {
        return get() != 0;
      }
 
-     void swap(scoped_ptr & b) // never throws
+     void swap(unique_ptr & b) // never throws
      {
        std::swap(ptr_, b.ptr_);
      }
+
    private:
-     scoped_ptr(const scoped_ptr&);
-     scoped_ptr& operator = (const scoped_ptr&);
+     unique_ptr(const unique_ptr&);
+     unique_ptr& operator = (const unique_ptr&);
 
      T* ptr_;
   };
 
-  template<typename T, typename Deleter>
-  void swap(scoped_ptr<T, Deleter> & a, scoped_ptr<T, Deleter> & b) // never throws
+
+  template<typename T>
+  rv<T>& move(T& p)
   {
-    return a.swap(b);
+    return static_cast< rv<T>& >(p);
   }
 
 } // namespace DCPS
 } // namespace OpenDDS
 
 OPENDDS_END_VERSIONED_NAMESPACE_DECL
 
-#endif /* end of include guard: SCOPED_PTR_H_18C6F30C */
+namespace std {
```

Comment:
Can this swap() be moved to the DCPS namespace?  Seee http://en.cppreference.com/w/cpp/concept/Swappable

Reply:
We are using namespace level swaps also for IDL to C++11, no problems there, specializing std::swap triggers a lot of compile warnings when you are using clang

Reply:
In ReceivedDataSample.inl:50, instead of directly calling std::swap(), use unqualified swap().  If necessary add "using std::swap;".